hocuspocus-extension-elasticsearch/src/elasticsearch.ts

```diff
@@ -1,25 +1,50 @@
-import { Database, DatabaseConfiguration } from '@hocuspocus/extension-database'
-import elasticsearch from '@elastic/elasticsearch'
+import {
+  Database,
+  DatabaseConfiguration,
+} from '@hocuspocus/extension-database';
+import elasticsearch from '@elastic/elasticsearch';
+
+export interface ElasticsearchConfiguration extends DatabaseConfiguration {
+  elasticsearchOpts?: elasticsearch.ClientOptions;
+  db_index?: string;
+}
 
 export class Elasticsearch extends Database {
-  db?: elasticsearch.Client
+  db?: elasticsearch.Client;
+  db_index: string;
 
-  configuration: DatabaseConfiguration = {
+  configuration: ElasticsearchConfiguration = {
     fetch: async ({ documentName }) => {
-      console.log('DB fetch')
-      const { body: docExist } = await this.db?.exists({ index: 'ydocs', id: documentName, type: 'doc' }) || {}
+      // console.log(`DB fetch ${documentName}`);
+      const { body: docExist } =
+        (await this.db?.exists({
```

Comment:
For not exist error, I think we can either check the status code returned by `this.db?.get`, or try-catch if it throws an error. In this case we don't need an extra `exist()` call to the database.